Google.PowerShell/Storage/GcsObject.cs

```diff
@@ -274,8 +304,69 @@ private class ParameterSetNames
         /// </summary>
         [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ContentsFromFile)]
         [Parameter(Mandatory = false, ParameterSetName = ParameterSetNames.ContentsFromString)]
+        [ValidateNotNullOrEmpty]
         public string ContentType { get; set; }
 
+        /// <summary>
```

Comment:
Do we need these extra parameters? The user can currently use the metadata to set these values.

Reply:
They are different than the custom metadata parameters and they can directly affect the storage object property (like cache control, etc.) so I think they should get their own parameters.